kubert/src/server.rs

```diff
@@ -180,8 +250,8 @@ async fn accept_loop<S, B>(
     tcp: TcpListener,
     drain: drain::Watch,
     service: S,
-    tls_key: TlsKeyPath,
-    tls_certs: TlsCertPath,
+    #[cfg(any(feature = "rustls-tls", feature = "boring-tls"))] tls_key: TlsKeyPath,
+    #[cfg(any(feature = "rustls-tls", feature = "boring-tls"))] tls_certs: TlsCertPath,
```

Comment:
I feel like we're taking on a lot of boilerplate to handle a case we don't actually care about supporting (disabling TLS). Is it possible to remove all of these cfgs?

Reply:
initially, i had the `server` module just require both the `server` feature _and_ one of the TLS features, so this stuff didn't have to be feature-flagged. 

i thought it was nicer to allow the `server` module to compile without a TLS implementation selected. but, the motivation was not that it was important to allow disabling server TLS, but that a crate whichisn't the top-level binary should be able to depend on APIs from the server module without picking which TLS impl is used. that way, the top-level binary can select the TLS implementation, but crates it depends on can also use APIs from the  `server` module. in order to allow that, the `server` module needs to compile without any TLS features selected.

if we don't like having to feature flag the TLS key/cert paths, we could just always have them be part of the struct, and just not use them when no TLS features are enabled. is that preferable?

Reply:
If you can make this behavior less verbose, I'm open to it.

Reply:
let me know what you think of 7a26377a2b509ad145374f8d14d1971d68af4f4e ?